src/EiffelEvents.RabbitMq.Client/RabbitMqEiffelClient.cs

```diff
@@ -87,18 +119,23 @@ public string Subscribe<T>(string serviceIdentifier, Action<T, ulong> callback)
                 _rabbitMqWrapper.CreateQueue(queueName, routingKey);
                 var subscriptionId = _rabbitMqWrapper.Consume(queueName, (_, eventArgs) =>
                     {
-                        var content = Encoding.UTF8.GetString(eventArgs.Body.Span);
-
-                        var typeObj = (T)Activator.CreateInstance(typeof(T));
-                        if (typeObj == null) return;
-
-                        var eiffelEvent = (T)typeObj.FromJson(content);
-                        ValidationHelper.ValidateEventVersion(eiffelEvent);
-
-                        callback(eiffelEvent, eventArgs.DeliveryTag);
+                        try
+                        {
+                            var content = Encoding.UTF8.GetString(eventArgs.Body.Span);
+
+                            var typeObj = (T)Activator.CreateInstance(typeof(T));
+                            if (typeObj == null) return;
+
+                            var result = ValidateEvent(validateOnSubscribe, typeObj, content);
```

Comment:
I suggets to rename this method to `TryDeseralizeEvent `(like the norm of `int.TryParse()` ) as it validates and deserialzes if possible.